libs/prelude/Prelude.idr

```diff
@@ -214,6 +214,30 @@ Enum Char where
 
   pred c = fromNat (pred (toNat c))
 
+Enum Bits8 where
+  toNat b   = fromIntegerNat (prim__zextB8_BigInt b)
```

Comment:
Thanks for the code.

I think the use of `prim__*` operators in real code should be avoided if possible.
Is it not possible to just use `fromInteger` for the `fromNat` one instead of `prim_trunc*`.

I don't know what the corresponding functions are in the other directions, but maybe you could make relevant cast instances.